### ISSUE TITLE
Shell out to php for modules/themes list

### ISSUE THREAD

**Issue by DrewML** [DrewML/baler#12]
We've currently hand-written all the code necessary to find all enabled modules/themes installed in a store.

This was done because we did not want to invoke `bin/magento` and incur the bootstrapping cost of Magento core every time `baler` is run.

However, @vinai pointed out to me today that it's just as fast (if not slightly faster) to invoke `autoload.php`, and then dump values from the `ComponentRegistrar`:

```sh
php -r "require 'vendor/autoload.php'; echo json_encode([(new \\Magento\\Framework\\Component\\ComponentRegistrar)->getPaths('theme'),(new \\Magento\\Framework\\Component\\ComponentRegistrar)->getPaths('module')]);"
```

Switching to this would resolve several open bugs, and allow us to delete several hundred lines of code. In addition, the maintenance burden will be significantly smaller when Magento core makes any changes to the theme/module lookup or registration rules.

**Comment by DrewML** [comment]
@friendscottn Can you run the command above and verify that all those different Mirasvit modules are getting picked up correctly? Can't imagine they're not, but verification is good :)

**Comment by DrewML** [comment]
Landed this in `master` in c0c3bac9
